pkg/internal/controller/controller.go

```diff
@@ -200,7 +200,7 @@ func (c *Controller[request]) Start(ctx context.Context) error {
 						sourceStartErrChan <- err
 						return
 					}
-					syncingSource, ok := watch.(source.SyncingSource)
+					syncingSource, ok := watch.(source.TypedSyncingSource[request])
```

Comment:
Oh wow, good catch. Mind adding a tiny test that validates that `WaitForSync` gets called for a `TypedSyncingSource`?

Reply:
added